src/api/scenarioData/dto/create-scenario.dto.ts

```diff
@@ -0,0 +1,32 @@
+import { IsNotEmpty, IsString, ValidateNested } from 'class-validator';
+import { Type } from 'class-transformer';
+import { Variable } from '../scenarioData.schema';
+
+export class VariableClass implements Variable {
+  name: string;
+  value: any;
+  type: string;
+}
+
+export class CreateScenarioDto {
+  @IsNotEmpty()
+  @IsString()
+  title: string;
+
+  @IsNotEmpty()
+  @IsString()
+  ruleID: string;
+
+  @IsNotEmpty()
+  @ValidateNested({ each: true })
+  @Type(() => VariableClass)
+  variables: VariableClass[];
+
+  @ValidateNested({ each: true })
+  @Type(() => VariableClass)
+  expectedResults: VariableClass[];
+
+  @IsNotEmpty()
+  @IsString()
+  goRulesJSONFilename: string;
```

Comment:
Do we need "goRuleJSONFilename" here? I see you have methods for getting scenarios by ruleId and JSON filename. Is there any reason we can't only use ruleID? Or is there a good reason to have both?

Reply:
I use the goRulesJSONFilename because the frontend so far directly has directly referenced the filename, while never using the ruleID, for many services. Both the decisions and documents services/controllers reference the file name instead of the ruleID, so I wanted to maintain this for ease of use and consistency.

Reply:
Yeah, I totally get the logic here since it does somewhat follow with what I've implemented (which was built that way to mimic the original GoRules cloud API). However, now that I'm seeing both the rule and the jsonfile getting stored in a separate db object and have duplicate methods for each of them, it's starting to feel superfluous. We should just pick one way and since we can always get the json file name from the rules id, I think it would make sense to just use the rules id and pair down some of the code.